File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int n = nums.size();

        int ans = 1000;
        int l = 0, r = n;

        while(l < r) {
            int m = l + (r - l) / 2;
            
            ans = min(ans, nums[m]);

            if( nums[m] > nums[r-1]){
                l = m+1;
            } else {
                r = m;
            } 
            
        }
        return ans;
        
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int n = nums.length;
        Map<Integer, Integer> count = new HashMap<>();
        count.put(0, 1);
        int ans = 0;
        int prefixSum = 0;
        for (int i = 0; i < n; i++) {
            prefixSum += nums[i];
            int sum = prefixSum - k;

            ans += count.getOrDefault(sum, 0);
            count.put(prefixSum, count.getOrDefault(prefixSum, 0) + 1);
        }

        // System.out.println(count.toString());
        return ans;
    }
}